app/assets/javascripts/oxalis/model/reducers/save_reducer.js

```diff
@@ -11,7 +11,7 @@ function SaveReducer(state: OxalisState, action: ActionType): OxalisState {
   switch (action.type) {
     case "PUSH_SAVE_QUEUE": {
       return update(state, {
-        save: { queue: { $push: action.items } },
+        save: { queue: { $push: action.items.length ? [action.items] : [] } },
```

Comment:
No you're not missing something and you're correct. However, there is a test that does the following:
```
  const items = [];
  const pushAction = SaveActions.pushSaveQueueAction(items);
  const newState = SaveReducer(initialState, pushAction);

  t.deepEqual(newState.save.queue, []);
```
So the test executes the pushSaveQueueAction without the check you mentioned and pushes an empty array.
One could argue that this test is good, because a developer could do the same and forget to do the `if (items.length > 0)` check and then we end up with an empty array in our save queue (that's what caused me to commit this change).

I have no strong opinion on whether we should include this change or not.